sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```diff
@@ -678,7 +683,8 @@ public void deleteDataset(String projectId, String datasetId)
         throws IOException, InterruptedException {
       checkNotNull(ref, "ref");
       if (executor == null) {
-        this.executor = options.as(GcsOptions.class).getExecutorService();
+        this.executor =
+            new BoundedExecutorService(options.as(GcsOptions.class).getExecutorService(), 3);
```

Comment:
Should we make this a pipelineOption?

Reply:
Do you mean the number of default threads or `BoundedExecutorService` itself?